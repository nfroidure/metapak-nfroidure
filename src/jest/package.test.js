'use strict';

const packageTransformer = require('./package');

describe('mocha', () => {
  describe('Package transformer', () => {
    it('should work with some files', () => {
      expect(
        packageTransformer({
          metapak: {
            data: {
              testsFiles: 'src/*.js src/**/*.js',
            },
          },
          scripts: {
            test: '',
          },
        }),
      ).toMatchSnapshot();
    });

    it('should work with child packages', () => {
      expect(
        packageTransformer({
          metapak: {
            data: {
              childPackage: true,
              files: 'src/*.js src/**/*.js',
            },
          },
        }),
      ).toMatchSnapshot();
    });
  });
});
