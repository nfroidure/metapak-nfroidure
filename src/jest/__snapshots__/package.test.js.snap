// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mocha Package transformer should work with child packages 1`] = `
Object {
  "devDependencies": Object {
    "jest": "^25.2.4",
  },
  "jest": Object {
    "coverageReporters": Array [
      "lcov",
    ],
    "roots": Array [
      "<rootDir>/src",
    ],
    "testPathIgnorePatterns": Array [
      "/node_modules/",
    ],
  },
  "metapak": Object {
    "data": Object {
      "childPackage": true,
      "files": "src/*.js src/**/*.js",
    },
  },
  "scripts": Object {
    "cover": "npm run jest -- --coverage",
    "jest": "NODE_ENV=test jest",
    "test": "npm run jest",
  },
}
`;

exports[`mocha Package transformer should work with some files 1`] = `
Object {
  "devDependencies": Object {
    "coveralls": "^3.0.11",
    "jest": "^25.2.4",
  },
  "greenkeeper": Object {
    "ignore": Array [
      "jest",
      "coveralls",
    ],
  },
  "jest": Object {
    "coverageReporters": Array [
      "lcov",
    ],
    "roots": Array [
      "<rootDir>/src",
    ],
    "testPathIgnorePatterns": Array [
      "/node_modules/",
    ],
  },
  "metapak": Object {
    "data": Object {
      "testsFiles": "src/*.js src/**/*.js",
    },
  },
  "scripts": Object {
    "cover": "npm run jest -- --coverage",
    "coveralls": "npm run cover && cat ./coverage/lcov.info | coveralls && rm -rf ./coverage",
    "jest": "NODE_ENV=test jest",
    "test": "npm run jest",
  },
}
`;
